intercept_server_test.go

```diff
@@ -304,12 +326,13 @@ func (s *testInterceptServerStream) SendMsg(m interface{}) error {
 //
 // testServer is not thread-safe.
 type testServer struct {
+	grpchantesting.UnimplementedTestServiceServer
 	code      codes.Code
 	resp      interface{}
 	respCount int
 	headers   metadata.MD
 	trailers  metadata.MD
-	calls     []interface{} // elements are either *unaryCall or *streamCall
```

Comment:
The unary and stream types were so darn similar... I'm not sure why I made them separate types to begin with. But this simplifies things (and makes the new `checkCalls` test helper way simpler).

Reply:
That cleanup seems like an easy 👍